applications/ConstitutiveModelsApplication/custom_models/elasticity_models/borja_model.cpp

```diff
@@ -0,0 +1,203 @@
+//
+//   Project Name:        KratosConstitutiveModelsApplication $
+//   Created by:          $Author:                  LMonforte $
+//   Last modified by:    $Co-Author:                         $
+//   Date:                $Date:                   April 2017 $
+//   Revision:            $Revision:                      0.0 $
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_models/elasticity_models/borja_model.hpp"
+
+
+namespace Kratos
+{
+
+   //******************************CONSTRUCTOR*******************************************
+   //************************************************************************************
+
+   BorjaModel::BorjaModel()
+      : HenckyHyperElasticModel()
+   {
+   }
+
+   //******************************COPY CONSTRUCTOR**************************************
+   //************************************************************************************
+
+   BorjaModel::BorjaModel(const BorjaModel& rOther)
+      : HenckyHyperElasticModel( rOther )
+   {
+   }
+
+   //********************************CLONE***********************************************
+   //************************************************************************************
+
+   ConstitutiveModel::Pointer BorjaModel::Clone() const
+   {
+      return ( BorjaModel::Pointer(new BorjaModel(*this)) );
+   }
+
+   //********************************ASSIGNMENT******************************************
+   //************************************************************************************
+   BorjaModel& BorjaModel::operator=(BorjaModel const& rOther)
+   {
+      HenckyHyperElasticModel::operator=(rOther);
+      return *this;
+   }
+
+   //*******************************DESTRUCTOR*******************************************
+   //************************************************************************************
+
+   BorjaModel::~BorjaModel()
+   {
+   }
+
+
+   // ****************************************************************
+   // ***************************************************************
+   // not in the correct place
+   void BorjaModel::SeparateVolumetricAndDeviatoricPart( const MatrixType& rA, double & rVolumetric, MatrixType& rDev, double & devNorm)
+   {
+      KRATOS_TRY
+
+      rVolumetric = 0;
+      for (unsigned int i = 0; i < 3; i++)
```

Comment:
It is a good practice to use ++i instead of i++ https://stackoverflow.com/questions/24853/what-is-the-difference-between-i-and-i